Jupyter/BacktestingFutureChainProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Logging;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace QuantConnect.Lean.Engine.DataFeeds
+{
+    /// <summary>
+    /// An implementation of <see cref="BacktestingFutureChainProvider"/> that reads the list of contracts from open interest zip data files
+    /// </summary>
+    public class BacktestingFutureChainProvider
```

Comment:
If we're following the pattern of `IOptionsChainProvider` then I think we'll want to add an `IFuturesChainProvider` so this type can derive from the interface.

Reply:
At team stand up meeting, we have decided to keep `BacktestingFutureChainProvider` in Jupyter and to not create the interface and have it all in `Common` project, since it is not as necessary as `IOptionsChainProvider`.

Reply:
*Until we can find a source for it in live trading.

Reply:
@AlexCatarino we should still add the interface even if no live implementation yet.